src/http/cookie.cr

```diff
@@ -81,7 +81,7 @@ module HTTP
       end
 
       CookieString    = /(?:^|; )#{Regex::CookiePair}/
-      SetCookieString = /^#{Regex::CookiePair}(?:; #{Regex::CookieAV})*$/
+      SetCookieString = /^#{Regex::CookiePair}(?:;\s?#{Regex::CookieAV})*$/
```

Comment:
Can anybody check the RFC whether zero or more or zero or one are allowed? In the case of zero or more this should probably use something like `{0,256}` or so rather than `?`

Reply:
According to [RFC 2965](https://tools.ietf.org/html/rfc2965#section-3.1), `white space is permitted between tokens.` It is not explicit if it meany "one" or "multiple" but I'd assume withour further specifictation it is to be understood as "any number".

Whitespaces are also allowed in other places, for example between attribute and `=` (this is explicitly mentioned in the RFC). Also the date formats like the one described by RFC 1123 actually allow arbitrary number of whitespace between the tokens. So there are several issues with this cookie string parser. This PR is probably good for a quick fix, but the entire parse will need to be refactored, probably using a custom parser instead of regular expressions.

Reply:
So the regex addition should be changed to `\s*`.

Reply:
I would put a specific limit like I suggested above, `*`  tends to be dangerous when fed untrusted input.

Reply:
so `\s{0,256}` ?

Reply:
Would be my suggestions, don't know what others think.

Reply:
@straight-shoota ? @RX14 ? @Sija ?

Reply:
I don't think it matters, really. If it was up to me, I'd choose `\s*` because it's simpler and the entire regex parser is not "safe" and needs an oberhaul anyway.

Reply:
We enforce a 16KiB length limit on parsed HTTP headers, so we should be fine.

Reply:
So... merge?

Reply:
No, change to `\s*`